capstone/src/main/webapp/scripts/search.js

```diff
@@ -1,52 +1,104 @@
 const searchInput = document.getElementById("name-search");
 const suggestionsPanel = document.getElementById("suggestions");
+const KeyCodes = {
+  UP: "38",
+  DOWN: "40",
+  ENTER: "13"
+};
 
-let names = [
-  { fullname: "Lucy Qu" },
-  { fullname: "Lucy Liu" },
-  { fullname: "Himani Yadav" },
-  { fullname: "Anika Bagga" },
-  { fullname: "Lucille Ball" }
-];
 let currentFocus = -1;
 
 function autocomplete() {
-  searchInput.addEventListener("keyup", filter);
   searchInput.addEventListener("keyup", checkKey);
 }
 
-function filter() {
-  const input = searchInput.value.toLowerCase();
-  suggestionsPanel.innerHTML = "";
-  const suggestions = names.filter(function(name) {
-    return name.fullname.toLowerCase().startsWith(input);
-  });
-  suggestions.forEach(suggested => appendSuggestion(suggested));
-  if (input === "") {
-    suggestionsPanel.innerHTML = "";
-  }
-}
-
 function checkKey(e) {
-  let list = document.getElementById("suggestions");
-  if (list) {
-    list = list.getElementsByTagName("div");
-  }
-  if (e.keyCode == "38") {
-    //if up arrow is pressed, highlight name
+  let suggestionsContainer = document.getElementById("suggestions");
+  let list = suggestionsContainer.getElementsByTagName("div");
+  if (e.keyCode == KeyCodes.UP) {
     currentFocus--;
     addActive(list);
-  } else if (e.keyCode == "40") {
-    //down arrow
+  } else if (e.keyCode == KeyCodes.DOWN) {
     currentFocus++;
     addActive(list);
-  } else if (e.keyCode == 13) {
-    //press enter
+  } else if (e.keyCode == KeyCodes.ENTER) {
+    // e.preventDefault();
+    getSearchResults(list);
+  } else {
+    getSuggestions();
+  }
+}
+
+function getSearchResults(list) {
+  let names = "";
+  if (list.length >= 5) {
+    //Gets top 5 suggested names
+    names = `${list[0].innerHTML},${list[1].innerHTML},${list[2].innerHTML},${list[3].innerHTML},${list[4].innerHTML}`;
```

Comment:
I don't want to have the results have too many names so I'm just limiting it to the top 5 names and finding results matching those top 5 names

Reply:
Okie dokie! A couple of things you might want to consider doing (as future changes, not in this PR):
1) If you're only using the top 5 names, then have the servlet only return the top 5 names.
2) Pagination for the names. Depending on how you do the sorting this might be less important feature but I can imagine a scenario in which a user searches for someone with a common name and their friend is not in the list of those 5 people. Pagination could be either serverside (you've already returned all the names so it's just using the pagination to display the next/previous set of 5) or clientside (how you did the comments in your portfolio)